### DISCUSSION TITLE
Java compatibility

### DISCUSSION THREAD

**Discussion by JG-QuarkNet** [eclipse-jdt/eclipse.jdt.core#1149] [Q&A]
I'm currently troubleshooting an issue with Apache Tomcat 9 on Ubuntu.  Tomcat 9 installs the package `libeclipse-jdt-core-java` as a dependency, which I believe corresponds to the code in this repository.  On Ubuntu 20.04, Tomcat 9 installs v 3.18.0+eclipse4.12-1 of this package, while on Ubuntu 22.04 it installs v 3.27.0+eclipse4.21-1.

Someone on the Tomcat users mailing list told me that JDT does not support Java 8 for JDT versions > 4.20, which would explain why I can get Tomcat 9 working with Java 8 on Ubuntu 20.04 but not on 22.04.  I assume that the JDT version is the one attached to the 'eclipse' portion of the above package versions, meaning that the package moves from JDT version 4.12 on Ubuntu 20.04 to JDT version 4.21 on Ubuntu 22.04.

If this is all true, it means that Tomcat 9 is incompatible with Java 8 when installed on Ubuntu 22.04.  Since Tomcat advertises v9 as fully compatible with Java 8, I suspect this was not intentional.  To make progress troubleshooting this issue, I'm asking the more knowledgeable people here to please confirm or contradict the following three statements:

1) The Ubuntu package `libeclipse-jdt-core-java` represents the code in this repo,
2) The bolded portion of "3.18.0+eclipse**4.12**-1" represents the JDT version of the package, and
3) That JDT versions greater than 4.20 are incompatible with Java 8.

Thank you for any help you can provide.

**Comment by trancexpress** [comment]
I believe jdt.core moved to Java 11 and dropped Java 8 support for Eclipse 4.21: https://www.eclipse.org/lists/eclipse-dev/msg11703.html
